src/python/grpcio/grpc/_common.py

```diff
@@ -97,3 +98,27 @@ def serialize(message, serializer):
 def deserialize(serialized_message, deserializer):
   return _transform(serialized_message, deserializer,
                     'Exception deserializing message!')
+
+
+class CleanupThread(threading.Thread):
+  """A wrapper around thread for custom join behavior.
+  
+  On Interpreter exit, Python will attempt to join outstanding threads
+  prior to garbage collection.  We may need to do additional cleanup, and
+  we acomplish this by wrapping the join() method.
+  """
+  
+  def __init__(self, group=None, target=None, join_target=None, name=None,
+               args=(), join_args=(), kwargs={}, join_kwargs={}):
+    """join_target() will be called with a timeout,
```

Comment:
Nearly without exception the doc string of an `__init__` method should be
```
Constructor.

Attributes:
  <attribute names and descriptions>
```
. No work of any consequence should be done in a constructor, and thus there should be nothing to say about it except to describe its parameters. See [the style guide](https://google.github.io/styleguide/pyguide.html) for [examples](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments).